CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -421,6 +421,90 @@ public function upgrade_4_7_24($rev) {
     $this->addTask('CRM-20958 - Add modified_date to civicrm_case', 'addColumn',
       'civicrm_case', 'modified_date', "timestamp NULL  DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT 'When was the case (or closely related entity) was created or modified or deleted.'");
 
+    $this->addTask('CRM-20958 - Suspend modification tracking during upgrade', 'toggleModificationTracking', FALSE);
+
+    list($minId, $maxId) = CRM_Core_DAO::executeQuery(
+      "SELECT coalesce(min(id),0), coalesce(max(id),0) FROM civicrm_activity"
+    )->getDatabaseResult()->fetchRow();
+    for ($startId = $minId; $startId <= $maxId; $startId += self::BATCH_SIZE) {
+      $endId = $startId + self::BATCH_SIZE - 1;
+      $vars = array(1 => array($startId, 'Int'), 2 => array($endId, 'Int'));
+
+      $title = sprintf('CRM-20958 - Compute civicrm_activity.created_date from civicrm_log (%d => %d)', $startId, $endId);
+      $sql = 'UPDATE civicrm_activity
+       SET created_date = (SELECT MIN(l.modified_date) FROM civicrm_log l WHERE l.entity_table ="civicrm_activity" AND civicrm_activity.id = l.entity_id)
```

Comment:
Confirmed:
 * Open a new browser. Login to `d46` as `admin`. Set your timezone to `America/New_York`. Edit contact 139. Observe "Change Log" has new entry with timestamp `July 26th, 2017 7:29 PM`.
 * Open a new browser. Login to `d46` as `demo`. Set your timezone to `America/Los_Angeles`. Edit contact 139. Observe "Change Log" has new entry with timestamp `July 26th, 2017 4:30 PM`. (In reality, this is 1 minute later.)

But the changelog (as viewed by either user) shows weird data: `admin`'s change actually happened **_1 minute before_** `demo`'s change, but it looks like it happened _**179 minutes after**_.

<img width="727" alt="screen shot 2017-07-26 at 4 34 29 pm" src="https://user-images.githubusercontent.com/1336047/28648107-53d619c8-7220-11e7-85a3-03efa4d30c2b.png">